packages/pyright-internal/src/languageService/hoverProvider.ts

```diff
@@ -482,10 +484,29 @@ export class HoverProvider {
     }
 
     private static _getTypeText(node: NameNode, evaluator: TypeEvaluator, expandTypeAlias = false): string {
-        const type = evaluator.getType(node) || UnknownType.create();
+        let type = evaluator.getType(node) || UnknownType.create();
+        type = this._limitOverloadBasedOnCall(node, evaluator, type);
```

Comment:
It would apply to variables and type aliases that evaluate to an overloaded callable. It would never apply to parameters or type parameters, but it doesn't hurt to have it in this common routine.